### ISSUE TITLE
iOS show not implemented

### ISSUE THREAD

**Issue by eborned** [driftyco/ionic-plugins-keyboard#46]
Is there a reason why show is not handles by the iOS plugin? Event if there is no way to open the keyboard, shouldn't the plugin still handle the function call with a warning message? At the moment, it results in ERROR: Method 'show' not defined in Plugin 'Keyboard'.

**Comment by wwweaponizer** [comment]
+1

**Comment by sandstrom** [comment]
+1

**Comment by tlancina** [comment]
It isn't implemented because you can just call focus() as long as you have `<preference name="KeyboardDisplayRequiresUserAction" value="false"/>` in your config.xml.

Closing for now, feel free to re-open if you have a scenario that requires show.

**Comment by theaccordance** [comment]
Can you please explain the logic behind not conforming a method across the platforms your framework supports?  As this thread stands, the decision to exclude the show method for iOS sounds more like laziness than it does technical difficulty or priority.

**Comment by tlancina** [comment]
@theaccordance  On Android you can show the keyboard regardless of input or focus:  [InputMethodManager#toggleSoftInput(int, int)](http://developer.android.com/reference/android/view/inputmethod/InputMethodManager.html#toggleSoftInput(int, int)).  This works nicely with a web view because you can call `focus()` on an input and then show the keyboard.

On iOS, the keyboard only appears when you have something to enter text into.  So calling `focus()` on an input will do it, or calling `becomeFirstResponder` on a native text field will also do it.  On Android, show is necessary because the keyboard doesn't always show when calling `focus()` on an input without user interaction.  This isn't the case on iOS, so it's unnecessary.

We could implement show by creating a hidden UITextField and calling `becomeFirstResponder` on it to force the keyboard open, but then it would steal focus from our input.  To bring the focus back, we have to call `focus()` on our input, which is what we could have done in the first place to bring the keyboard up.

tl;dr Android and iOS behave differently so there's no way to get the same behavior which is unnecessary on iOS anyways.

**Comment by YamanSehzade** [comment]
Hi @tlancina ,
Is any way to show keyboard on IOS? Maybe, something changed in two years.
